webapp/app/routes.py

```diff
@@ -316,3 +317,16 @@ def error_500(error):
         current_app.logger.error(
             'An uncaught error occurred', exc_info=error.original_exception)
         return render_template('error/500.html', header_title=_('500.header-title'), js_needed=False), 500
+
+
+@non_production_environment_required
+def register_testing_request_handlers(app):
```

Comment:
Hmm. Now that I see it in action, I'm no longer sure I like that we made this into a decorator: it implies that this is something we want to do generally and invite re-use. The opposite is the case: we're poking a very small hole into the application with little intention to do so ever again (though who can predict the future).

Have you considered and ruled out the simpler solution with just a guard clause?

```suggestion
def register_testing_request_handlers(app):
    if app.env == 'production':
        return
```

Reply:
Great idea.